### PR TITLE
[WIP] feat: Setup deployment microservice

### PR DESCRIPTION
**Pull request by gbanu**

### CODE REVIEW

File: server/deployment-service/build.gradle
Comment (thielpa) [inline server/deployment-service/build.gradle:1]: proposal: Seems like there are different spacings used in this file. I would propose to use a linter e.g. checkstyle also for the server part (of course in a separate issue)

File: server/deployment-service/.gitignore
Comment (thielpa) [inline server/deployment-service/.gitignore:1]: discussion: can we share some of them in a global .gitignore in the server folder for all microservices? Not sure though, but I do not like the code duplication

File: server/deployment-service/src/main/java/de/tum/cit/aet/helios/deployment/environments/EnvironmentInfoDTO.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import de.tum.cit.aet.helios.deployment.deployments.DeploymentDTO;
+import de.tum.cit.aet.helios.deployment.environments.EnvironmentInfo.Status;
+import de.tum.cit.aet.helios.deployment.installed_apps.InstalledAppDTO;
+
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record EnvironmentInfoDTO(
+        @NonNull Long id,
+        @NonNull String name,
+        @NonNull String description,
```
Comment (thielpa) [inline server/deployment-service/src/main/java/de/tum/cit/aet/helios/deployment/environments/EnvironmentInfoDTO.java:19]: question: Shouldn't the description be nullable (or empty string)?

File: server/deployment-service/src/main/java/de/tum/cit/aet/helios/deployment/environments/EnvironmentInfoDTO.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        @NonNull String url,
+        @NonNull Status status,
+        @NonNull Long createdByUserId,
+        OffsetDateTime createdAt,
+        List<InstalledAppDTO> installedApps,
```
Comment (thielpa) [inline server/deployment-service/src/main/java/de/tum/cit/aet/helios/deployment/environments/EnvironmentInfoDTO.java:24]: question: Should `installedApps` be nullable?

File: server/deployment-service/src/main/java/de/tum/cit/aet/helios/deployment/environments/EnvironmentInfoDTO.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                environment.getName(),
+                environment.getDescription(),
+                environment.getUrl(),
+                environment.getStatus(),
+                environment.getCreatedByUserId(),
+                environment.getCreatedAt(),
+                environment.getInstalledApps()
+                        .stream()
+                        .map(InstalledAppDTO::toInstalledAppDTO)
+                        .sorted()
```
Comment (thielpa) [inline server/deployment-service/src/main/java/de/tum/cit/aet/helios/deployment/environments/EnvironmentInfoDTO.java:41]: question: Is this sorting by name?

**Review by thielpa — APPROVED** [review]
LGTM, good work in my opinion 🚀

**Comment by egekocabas** [comment]
I close this PR and will create another one to not to send notification emails in every commit 😅

**Pull request closed without merging**